src/main/java/seedu/duke/Api.java

```diff
@@ -0,0 +1,42 @@
+package seedu.duke;
+
+import com.google.gson.Gson;
+import netscape.javascript.JSObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.http.HttpClient;
+import java.util.List;
+
+public class Api {
+
+    String urlString = "https://api.nusmods.com/v2/2020-2021/moduleList.json";
+
+    public Api() {
+    }
+
+    public Module[] getAllModules() throws IOException {
+        URL url = new URL(urlString);
+        URLConnection conn = url.openConnection();
+        InputStream is = conn.getInputStream();
+        BufferedReader bR = new BufferedReader(  new InputStreamReader(is));
```

Comment:
The gradle build test for ubuntu detect a error for bR (not sure why). Also for the `new BufferedReader(  new InputStreamReader(is));`, they throw error for having a space between the '(' and "new"